File: appjam/src/main/java/com/example/appjambackend/domain/user/entity/refresh/RefreshToken.java
package com.example.appjambackend.domain.user.entity.refresh;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

import java.io.Serializable;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@RedisHash
public class RefreshToken implements Serializable {

    @Id
    private String userid;

    @Indexed
    private String refreshToken;

    @TimeToLive
    private Long refreshExpiration;

    @Builder
    public RefreshToken(String userid, String refreshToken, Long refreshExpiration) {
        this.userid = userid;
        this.refreshToken = refreshToken;
        this.refreshExpiration = refreshExpiration;
    }

    public String reissue(String refreshToken, Long refreshExpiration) {
        this.refreshToken = refreshToken;
        this.refreshExpiration = refreshExpiration;
        return this.refreshToken;
    }
}

File: src/main/kotlin/com/example/mitalk/domain/customer/domain/entity/RefreshToken.kt
package com.example.mitalk.domain.customer.domain.entity

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed
import java.util.*

@RedisHash(value = "refreshToken", timeToLive = 60L * 60 * 24 * 7)
class RefreshToken(
        @Id
        @Indexed
        val userId: UUID? = null,
        @Indexed
        val token: String,
)